### DISCUSSION TITLE
Mojo Environment - Validate / Modify / Migrate to run w/ Mojo

### DISCUSSION THREAD

**Discussion by mavrukin** [pallets/flask#5302] [Ideas]
Mojo is now officially available for Mac, has been on Linux for ~a month.  Based on my personal experience, it delivers a very significant performance boost.  The proposal here is to make the necessary changes to Flask in order to take advantage of Mojo to its outmost.  

https://www.modular.com/blog/mojo-is-now-available-on-mac?utm_medium=email&_hsmi=279018192&_hsenc=p2ANqtz-9b34JWDfttyOU8NOU49xr8M-aXxkVoSk9LvnnaRSFOETY9TWysOdaxa076k9FOU0e13H2Q6EgffPoJpE5SMfre39KxgA&utm_content=279018192&utm_source=hs_email

I'm willing to dig into this and see what is necessary and own this through.

**Comment by RNRetailer** [comment]
Was Flask ever ported to Mojo?

Thanks

**Reply by CheeseCake87** [reply]
Nope. Not that I'm aware of. 

I know that Mojo is able to import Python libs but I'm not sure on the mechanisms behind this; but judging by the amount of time that has passed since the OP, I'd imagine it can't be that easy.

You're likely better calling Mojo from Flask in some way, than trying to fight with running Flask in Mojo. 

From the Mojo docs it looks like it prefers self contained Python libraries...

**Reply by mavrukin** [reply]
I'm not sure that statement is correct... the best way of thinking of Mojo is as another execution environment that takes advantage of the type information on the Python side in order to generate much more efficient code. 

If Flask is made to work inside of Mojo, which IS now available for Mac, there is likely to be a significant increase in performance.

**Reply by CheeseCake87** [reply]
> I'm not sure that statement is correct... the best way of thinking of Mojo is as another execution environment that takes advantage of the type information on the Python side in order to generate much more efficient code.
> 
> If Flask is made to work inside of Mojo, which IS now available for Mac, there is likely to be a significant increase in performance.

Are you any further forward with a possible solution of integration?
